Packet++/header/InfiniBandLayer.h

```diff
@@ -0,0 +1,288 @@
+		uint8_t getOpcode() const;
+
+		/**
+		 * Set operation code
+		 * @param[in] opcode The opcode to set
+		 */
+		void setOpcode(uint8_t opcode) const;
+
+		/**
+		 * @return solicited event that the responder shall invoke the CQ event handler
+		 */
+		uint8_t getSe() const;
+
+		/**
+		 * Set solicited event
+		 * @param[in] se The solicited event to set
+		 */
+		void setSe(int se) const;
+
+		/**
+		 * @return migreq which used to communicate migration state
+		 */
+		uint8_t getMig() const;
+
+		/**
+		 * Set migreq
+		 * @param[in] mig The migration state to set. If set to one, indicates the connection or EE context has been
+		 * migrated; if set to zero, it means there is no change in the current migration state.
+		 */
+		void setMig(uint8_t mig) const;
+
+		/**
+		 * @return PadCount which Packet payloads are sent as a multiple of 4-byte quantities.
+		 * Pad count indicates the number of pad bytes - 0 to 3 - that are appended to the packetpayload.
+		 * Pads are used to “stretch” the payload (payloads may be zero or more bytes in length) to be a multiple of 4
+		 * bytes
+		 */
+		uint8_t getPad() const;
+
+		/**
+		 * Set PadCount
+		 * @param[in] pad The PadCount to set
+		 */
+		void setPad(uint8_t pad) const;
+
+		/**
+		 * @return Transport Header Version that specifies the version of the IBA Transport used for this packet
+		 */
+		uint8_t getTver() const;
+
+		/**
+		 * Set Transport Header Version
+		 * @param[in] tvr The transport header version to set
+		 */
+		void setTver(uint8_t tver) const;
+
+		/**
+		 * @return partition key identifying the partition
+		 * that the destination QP (RC, UC, UD, XRC) or EE Context (RD) is a member.
+		 */
+		uint16_t getPkey() const;
+
+		/**
+		 * Set partition key
+		 * @param[in] pkey The partition key to set
+		 */
+		void setPkey(uint16_t pkey) const;
+
+		/**
+		 * @return destination queue pair (QP) identifier
+		 */
+		uint32_t getQpn() const;
+
+		/**
+		 * Set Queue Pair Number
+		 * @param[in] qpn The queue pair number to set
+		 */
+		void setQpn(uint32_t qpn) const;
+
+		/**
+		 * @return FECN
+		 * F (FECN): 0 indicates that a FECN indication was not received.
+		 * 1 indicates that the packet went through a point of congestion
+		 */
+		int getFecn() const;
+
+		/**
+		 * Set Fecn
+		 * @param[in] fecn The FECN to set
+		 */
+		void setfecn(int fecn) const;
+
+		/**
+		 * @return BECN
+		 * B (BECN): 0 the packet did not go through a point of congestion or went
+		 * through a point of congestion but was not marked. 1 indicates that the
+		 * packet indicated by this header was subject to forward congestion. The B
+		 * bit is set in an ACK or CN BTH
+		 */
+		int getBecn() const;
+
+		/**
+		 * Set BECN
+		 * @param[in] becn The BECN to set
+		 */
+		void setbecn(int becn) const;
+
+		/**
+		 * @return Reserved (variant) - 6 bits. Transmitted as 0, ignored on receive.
+		 */
+		uint8_t getResv6a() const;
+
+		/**
+		 * Set Reserved 6 bits
+		 */
+		void setResv6a() const;
```

Comment:
These are just reserved bits, do we need getters and setters for them?

Reply:
```
static inline u8 __bth_resv6a(void *arg)
{
	struct rxe_bth *bth = arg;

	return (BTH_RESV6A_MASK & be32_to_cpu(bth->qpn)) >> 24;
}

static inline void __bth_set_resv6a(void *arg)
{
	struct rxe_bth *bth = arg;

	bth->qpn &= cpu_to_be32(~BTH_RESV6A_MASK);
}
```
If we don't need it, I can del these code. However, the linux kernle reserve them.

Reply:
I don't have a strong opinion, you seem to know the protocol better than me. Do you think users will need this getter and setter? If you think yes then keep it, otherwise remove it

Reply:
I think we might won't use the func. I'll del them.

Reply:
del